storage/file_persister.go

```diff
@@ -43,3 +48,142 @@ func (l *LocalFilePersister) Persist(path string, data io.Reader) (err error) {
 
 	return nil
 }
+
+// RemoteFilePersister is to be used when files created by the browser module need
+// to be uploaded to a remote location. This uses a preSignedURLGetterURL to
+// retrieve one pre-signed URL. The pre-signed url is used to upload the file
+// to the remote location.
+type RemoteFilePersister struct {
+	preSignedURLGetterURL string
+	headers               map[string]string
+	basePath              string
+}
+
+// Persist will upload the contents of data to a remote location.
+func (r *RemoteFilePersister) Persist(ctx context.Context, path string, data io.Reader) (err error) {
+	pURL, err := r.getPreSignedURL(ctx, path)
+	if err != nil {
+		return fmt.Errorf("get presigned url: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPut, pURL, data)
+	if err != nil {
+		return fmt.Errorf("create upload request: %w", err)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("perform upload request: %w", err)
+	}
+	defer func() {
+		if _, cerr := io.Copy(io.Discard, resp.Body); cerr != nil && err == nil {
+			err = fmt.Errorf("drain upload response body: %w", cerr)
+		}
```

Comment:
This part might go into the `if err != nil` above. No particular reason I can see for this to be in a `defer`.

Reply:
Yep, good spot, resolved in 3c3b6fb879480090dc573d80f89aff295a25ec64

Reply:
Do we still need `cerr` there?

Reply:
Resolved in 66062104624c5f6267ec6dc986f5b9ce303f81bb